src/extensions/replay/sessionrecording.ts

```diff
@@ -786,11 +768,18 @@ export class SessionRecording {
         const event = truncateLargeConsoleLogs(throttledEvent)
         const size = JSON.stringify(event).length
 
+        // we need to ensure we rotate the session id
+        // before recording that the session has seen a full snapshot
         this._updateWindowAndSessionIds(event)
 
-        // allow custom events even when idle
-        if (this.isIdle && event.type !== EventType.Custom) {
-            // When in an idle state we keep recording, but don't capture the events
+        if (rawEvent.type === EventType.FullSnapshot) {
+            this.lastSessionFullSnapshot = this.sessionId
```

Comment:
We should possibly rename this if we're storing the `sessionId`, I assumed it was the full snapshot